### DISCUSSION TITLE
HTML tables with markdown content

### DISCUSSION THREAD

**Discussion by hellt** [squidfunk/mkdocs-material#6658] [Q&A] [answered]
Hi, I am trying to create some tables with nested markdown content, and original markdown syntax won't cut it.

I tried to use the raw HTML, but it seems that md_in_html doesn't quite work there as I expected it to:

````html
<table>
  <thead>
    <tr>
      <th markdown>
**Header1**
      </th>
      <th markdown>
**header2**
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td markdown>
```
some code
```
      </td>
      <td markdown>

some `markdown` **text**
      </td>
    </tr>
  </tbody>
</table>
````

render as 

<img width="332" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/5679861/305a8e5b-dfe9-4fe8-b200-f97fd17657f0">

I can use Blocks for that, but was curious if there is a way with native html

**Comment by alexvoss** [comment]
Hi, I may be wrong but I think you need the `markdown` attribute on all the HTML tags, so on `<table>` and `<tbody>`, for example.

**Reply by hellt** [reply]
I tried I think all combinations, none of them worked for me

**Reply by facelessuser** [reply, marked as answer]
`md_in_html` requires you to not indent the block elements:

````py
import markdown

MD = """
<table markdown>
<thead markdown>
<tr markdown>
<th markdown>
**Header1**
</th>
<th markdown>
**header2**
</th>
</tr>
</thead>
<tbody markdown>
<tr markdown>
<td markdown>
```
some code
```
</td>
<td markdown>
some `markdown` **text**
</td>
</tr>
</tbody>
</table>
"""

html = markdown.markdown(
    MD,
    extensions=['md_in_html'],
)

print(html)
````

```html
<table>
<thead>
<tr>
<th>
<strong>Header1</strong>
</th>
<th>
<strong>header2</strong>
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<code>some code</code>
</td>
<td>
some <code>markdown</code> <strong>text</strong>
</td>
</tr>
</tbody>
</table>
```

**Reply by hellt** [reply]
Got it, thanks again, @facelessuser 
This defeats the purpose as nesting gets completely lost.

Ok, I think I will have to recall how to python and write the blocks extension :D

```
/// thead
heading 1
///
/// thead
heading 2
///

//// trow
/// tcell
cell1-row1
///

/// tcell
cell2-row1
///
////

//// trow
/// tcell
cell1-row2
///

/// tcell
cell2-row2
///
////
```

**Reply by facelessuser** [reply]
Yep, just more stating how it works 🙂.

**Reply by alexvoss** [reply]
The rule that block elements must not be indented does not seem to be documented? I would expect it under [Nesting](https://python-markdown.github.io/extensions/md_in_html/#nesting) in the Python Markdown documentation?

**Reply by facelessuser** [reply]
Not it is not mentioned, but Python Markdown handles Markdown blocks, you can't have paragraphs randomly indented or they become code blocks etc. The HTML needs to conform to these rules.

The other downside is that HTML must be at the root level. You can't have block HTML under lists etc. or they likely won't always work as expected. While I'm generally fine with the indentation rule, the HTML must be at root level rule annoys me, but it is what it is. If I'm nesting under lists, I use `pymdownx.blocks.html`.

**Reply by alexvoss** [reply]
Yep. That is why I *love* the block syntax. 

Will see if people will accept a small PR over at Python Markdown central to clarify this in the documentation. Would "the HTML needs to be indented according to Markdown rules" be sufficient and not misleading?
